app/models/subnet.rb

```diff
@@ -113,8 +113,28 @@ def model_name
 
   delegate :supports_ipam_mode?, :supported_ipam_modes, :show_mask?, to: 'self.class'
 
+  apipie :class, "A class representing #{model_name.human} object" do
+    sections only: %w[all additional]
+    prop_group :basic_model_props, ApplicationRecord
+    property :boot_mode, String, desc: 'Returns the default boot mode for interfaces assigned to this subnet, applied to hosts from provisioning templates, e.g static or DHCP'
+    property :cidr, Integer, desc: 'Returns the network prefix or nil in case of Invalid Error'
+    property :description, String, desc: 'Returns the subnet description'
+    property :dhcp?, one_of: [true, false], desc: 'Returns true if this subnet supports the DHCP IPAM mode, false otherwise'
+    property :dhcp_boot_mode?, one_of: [true, false], desc: 'Returns true if the subnet boot mode is DHCP, false otherwise'
+    property :dns_primary, String, desc: 'Returns the primary DNS server address, e.g. 192.168.122.1'
+    property :dns_secondary, String, desc: 'Returns the secondary DNS server address'
+    property :dns_servers, Array, desc: 'Returns an array of the DNS servers address'
+    property :gateway, String, desc: 'Returns the gateway address, e.g. 192.168.122.1'
+    property :has_vlanid?, one_of: [true, false], desc: 'Returns true is the subnet has a VLAN ID, false otherwise'
```

Comment:
to_label, cidr, dhcp?,  dhcp_boot_mode? has_vlanid?, dns_servers doc shouln't be documented as part of the apipie methods?

Reply:
@shiramax, why should they? They are methods that can be called on a subnet object, they don't have any parameters, and they just return a value. They could be documented as part of the apipie methods (although not sure what do you mean exactly), but that would lead to redundant lines.

Reply:
@ofedoren ok sounds ok.